src/net.c

```diff
@@ -330,7 +330,7 @@ static int submit_send(struct super_block *sb,
 	    WARN_ON_ONCE(id == 0 && (flags & SCOUTFS_NET_FLAG_RESPONSE)))
 		return -EINVAL;
 
-	msend = kmalloc(offsetof(struct message_send,
+	msend = kzalloc(offsetof(struct message_send,
```

Comment:
We don't need to zero the whole allocation. Almost all of it is immediately overwritten by data from the sending caller or the receiving socket. Just set the net_header padding fields on the send side?